code/modules/bitrunning/antagonists/_parent.dm

```diff
@@ -60,13 +60,36 @@
 
 	return TRUE
 
+
 /// Sets up the agent so that they look like cyber police && don't have an account ID
-/datum/antagonist/bitrunning_glitch/proc/convert_agent(mob/living/carbon/human/player, datum/outfit/agent_outfit)
+/datum/antagonist/bitrunning_glitch/proc/convert_agent()
+	if(!ishuman(owner.current))
+		return
+
+	var/mob/living/carbon/human/player = owner.current
+
 	player.set_service_style()
-	player.equipOutfit(agent_outfit)
-	player.fully_replace_character_name(player.name, pick(GLOB.cyberauth_names))
+	player.equipOutfit(preview_outfit)
+
+	var/chosen_name = pick(GLOB.cyberauth_names)
+
+	player.fully_replace_character_name(player.name, chosen_name)
+	fix_agent_id()
+
+
```

Comment:
```suggestion
```

Reply:
I'm starting to give all of my procs two newlines